app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java

```diff
@@ -165,78 +166,90 @@ public static boolean isGithubApk(@NonNull final App app) {
         return getCertificateSHA1Fingerprint(app).equals(GITHUB_APK_SHA1);
     }
 
-    @Nullable
-    public static Disposable checkNewVersion(@NonNull final App app) {
+    private void checkNewVersion() throws IOException, ReCaptchaException {
+        final App app = App.getApp();
+
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(app);
         final NewVersionManager manager = new NewVersionManager();
 
         // Check if user has enabled/disabled update checking
         // and if the current apk is a github one or not.
         if (!prefs.getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
-            return null;
+            return;
         }
 
         // Check if the last request has happened a certain time ago
         // to reduce the number of API requests.
         final long expiry = prefs.getLong(app.getString(R.string.update_expiry_key), 0);
         if (!manager.isExpired(expiry)) {
-            return null;
+            return;
+        }
+
+        final Response response = getLatestNewPipeVersionsData();
```

Comment:
You shouldn't make requests in the main thread. A Disposable was used before to check for the new app version while starting the app, so as to prevent slowdowns

Reply:
> You shouldn't make requests in the main thread. A Disposable was used before to check for the new app version while starting the app, so as to prevent slowdowns

That is not true. I am using an IntentService that will run in a background thread.

Reply:
Oh, right, sorry